File: servico-musica-java/java.rest/src/main/java/servico/java/rest/entity/Musica.java
package servico.java.rest.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.util.List;

@Entity
public class Musica {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nome;
    private String artista;

    @ManyToMany(mappedBy = "musicas")
    @JsonIgnore
    private List<Playlist> playlists;

    protected Musica() {
    }

    public Musica(String nome, String artista) {
        this.nome = nome;
        this.artista = artista;
    }

    public long getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public String getArtista() {
        return artista;
    }

    public List<Playlist> getPlaylists() {
        return playlists;
    }
}

File: servico-musica-kotlin/kotlin.graphql/src/main/kotlin/servico/kotlin/graphql/entity/Musica.kt
package servico.kotlin.graphql.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
class Musica {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0
    var nome: String? = null
        private set
    var artista: String? = null
        private set

    @ManyToMany(mappedBy = "musicas")
    @JsonIgnore
    val playlists: MutableList<Playlist?>? = null

    protected constructor()

    constructor(nome: String?, artista: String?) {
        this.nome = nome
        this.artista = artista
    }
}